models/articles.model.js

```diff
@@ -1,5 +1,39 @@
 const db = require("../db/connection");
 
+exports.fetchArticles = (topic) => {
+  if (topic) {
+    return db
+      .query(
+        `SELECT articles.*, CAST (COUNT(comment_id) AS INT) AS comment_count 
+      FROM articles 
+      JOIN comments ON articles.article_id = comments.article_id WHERE topic = $1 GROUP BY articles.article_id ORDER BY article_id ASC;`,
+        [topic]
+      )
+      .then((data) => {
+        if (!data.rows.length) {
+          return Promise.reject({ status: 404, message: "Article not found" });
```

Comment:
We should be checking if the topic exists. If a topic exists and there are no articles for it, I would argue this could be viewed as a perfectly acceptable request. It would still however return nothing on the rows. However if the topic does not exist, then this could be viewed as a 404: not found. We could do this check if the length of what comes back is 0.